scripts/ci/publish

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+: ${SNAP_CI?This script is designed to be run within SnapCI. Aborting in a cowardly fashion.}
+
+
+# Login to NPM
+local npmrc_contents="//registry.npmjs.org/:_authToken=${NPM_TOKEN}"
```

Comment:
I ran into issues with Snap when using this technique. I can't remember the exact details, but I ended up resolving it by adding ~/.npmrc as a secure file in the Snap UI.